templates/console/source/lambda/config-version/cf_config_version_manager_graphql/cf_config_version_manager_graphql.py

```diff
@@ -31,66 +26,65 @@
 
 @app.resolver(type_name="Query", field_name="diffCloudfrontConfig")
 def manager_version_diff(distribution_id: str = "", version1: str = "", version2: str = ""):
-    dist_id = distribution_id
-    version_1 = version1
-    version_2 = version2
+    s3_client = boto3.client('s3')
+    diff_content = get_version_diff(s3_client, distribution_id, version1, version2)
+    return diff_content
+
 
+def get_version_diff(s3_client, dist_id, version1, version2):
     # get specific cloudfront distributions version info
     ddb_client = boto3.resource('dynamodb')
     ddb_table = ddb_client.Table(DDB_VERSION_TABLE_NAME)
-
     response = ddb_table.get_item(
         Key={
             "distributionId": dist_id,
-            "versionId": int(version_1)
+            "versionId": int(version1)
         })
     data = response['Item']
-
     s3_bucket = data['s3_bucket']
     s3_key1 = data['s3_key']
-
     response = ddb_table.get_item(
         Key={
             "distributionId": dist_id,
-            "versionId": int(version_2)
+            "versionId": int(version2)
         })
     data = response['Item']
     s3_key2 = data['s3_key']
-
-    s3_client = boto3.client('s3')
-    local_config_file_name_version1 = '/tmp/' + dist_id + "_" + version_1 + ".json"
-    local_config_file_name_version2 = '/tmp/' + dist_id + "_" + version_2 + ".json"
+    # s3_client = boto3.client('s3')
+    local_config_file_name_version1 = '/tmp/' + dist_id + "_" + version1 + ".json"
+    local_config_file_name_version2 = '/tmp/' + dist_id + "_" + version2 + ".json"
     s3_client.download_file(s3_bucket, s3_key1, local_config_file_name_version1)
     s3_client.download_file(s3_bucket, s3_key2, local_config_file_name_version2)
-
     # compare the two files
     cmd = ['git', 'diff', '--no-index', local_config_file_name_version1, local_config_file_name_version2,
            '>/tmp/diff.txt', ';', 'exit 0']
-
     shell_cmd = ' '.join(cmd)
     log.info(shell_cmd)
-
     output = subprocess.check_output(shell_cmd, shell=True)
-
     diff_file = open("/tmp/diff.txt", "r")
-
     diff_content = diff_file.read()
-
     diff_file.close()
-
     return diff_content
 
 
 @app.resolver(type_name="Query", field_name="diffCloudfrontConfigSnapshot")
 def manager_snapshot_diff(distribution_id: str = "", snapshot1: str = "", snapshot2: str = ""):
+    if distribution_id == "" or snapshot1 == "" or snapshot2 == "":
+        raise Exception("Snapshot name can not be empty")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.